rust/src/extensions.rs

```diff
@@ -185,6 +200,32 @@ extensions:
         assert!(extensions.get_repos() == vec!["my-repo"]);
         assert!(extensions.get_os_extension_packages() == vec!["bazboo"]);
         assert!(extensions.get_development_packages().is_empty());
+
+        let src_tf =
+            crate::Treefile::new_from_string(crate::treefile::tests::VALID_PRELUDE_JS).unwrap();
+        let tf = extensions.generate_treefile(&src_tf).unwrap();
+
+        assert_ne!(src_tf.parsed, tf.parsed);
+
+        let allrepos: Vec<_> = tf
+            .parsed
+            .repos
+            .as_ref()
+            .unwrap()
+            .iter()
+            .map(|s| s.as_str())
+            .collect();
+        // rpm-md repos are unioned
+        assert_eq!(allrepos, vec!["baserepo", "my-repo"]);
```

Comment:
Looks like this might be cleaner with https://docs.rs/itertools/0.10.1/itertools/fn.assert_equal.html